src/Android/Services/DeviceActionService.cs

```diff
@@ -108,22 +112,102 @@ public async Task ShowLoadingAsync(string text)
+                .SetCancelable(false)
+                .Create();
             _progressDialog.Show();
         }
 
         public Task HideLoadingAsync()
         {
-            if (_progressDialog != null)
+            // Based on https://github.com/redth-org/AndHUD/blob/master/AndHUD/AndHUD.cs
+            lock (_progressDialogLock)
+            {
+                if (_progressDialog is null)
+                {
+                    return Task.CompletedTask;
+                }
+
+                void actionDismiss()
+                {
+                    try
+                    {
+                        if (IsAlive(_progressDialog) && IsAlive(_progressDialog.Window))
+                        {
+                            _progressDialog.Hide();
+                            _progressDialog.Dismiss();
+                        }
+                    }
+                    catch
+                    {
+                        // ignore
+                    }
+
+                    _progressDialog = null;
+                }
+
+                // First try the SynchronizationContext
+                if (Application.SynchronizationContext != null)
+                {
+                    Application.SynchronizationContext.Send(state => actionDismiss(), null);
+                    return Task.CompletedTask;
+                }
+
+                // Otherwise try OwnerActivity on dialog
+                var ownerActivity = _progressDialog?.OwnerActivity;
+                if (IsAlive(ownerActivity))
+                {
+                    ownerActivity.RunOnUiThread(actionDismiss);
+                    return Task.CompletedTask;
+                }
+
+                // Otherwise try get it from the Window Context
+                if (_progressDialog?.Window?.Context is Activity windowActivity && IsAlive(windowActivity))
+                {
+                    windowActivity.RunOnUiThread(actionDismiss);
+                    return Task.CompletedTask;
+                }
+
+                // Finally if all else fails, let's see if current activity is MainActivity
+                if (CrossCurrentActivity.Current.Activity is MainActivity activity && IsAlive(activity))
+                {
+                    activity.RunOnUiThread(actionDismiss);
+                    return Task.CompletedTask;
+                }
+
+                return Task.CompletedTask;
+            }
+        }
+
+        bool IsAlive(Java.Lang.Object @object)
+        {
+            if (@object == null)
+                return false;
+
+            if (@object.Handle == IntPtr.Zero)
+                return false;
+
+            if (@object is Activity activity)
             {
-                _progressDialog.Dismiss();
-                _progressDialog.Dispose();
-                _progressDialog = null;
+                if (activity.IsFinishing)
+                    return false;
+
+                if (Build.VERSION.SdkInt >= BuildVersionCodes.JellyBeanMr1 &&
```

Comment:
Fwiw we only support 5.x +

Reply:
I forgot to remove that sorry, making the change asap